core/modules/layout/layout.module

```diff
@@ -1604,7 +1606,9 @@ function layout_get_layout_template_info($template_name = NULL, $rebuild = FALSE
       $init[$name]['title'] = $init[$name]['name'];
       $init[$name]['name'] = $name;
     }
-    $info = _layout_get_all_info('layout', $init);
+    $info = _layout_get_all_info('layout_template', $init);
+    // Account for implementations of the deprecated hook_layout_info().
+    $info = _layout_get_all_info('layout', $info);
```

Comment:
Not sure what's happening here 🤔

Reply:
The 2nd Parameter of this function is any existing template info, so the first call `$init` is existing info and then in the second call `$info` is now that existing info.